src/Umbraco.Core/Models/ILanguage.cs

```diff
@@ -33,5 +33,11 @@ public interface ILanguage : IEntity, IRememberBeingDirty
         /// If true, a variant node cannot be published unless this language variant is created
         /// </summary>
         bool Mandatory { get; set; }
+
+        /// <summary>
+        /// Defines the fallback language that can be used in multi-lingual scenarios to provide
+        /// content if the requested language does not have it published.
+        /// </summary>
+        ILanguage FallbackLanguage { get; set; }
```

Comment:
We should make this an `int?` and not have a direct reference to the other ILanguage. This will also simplify a few things with this implementation. The primary reason is because we are slowly trying to reduce the object graphs that get cached. All languages get cached and this will mean there's actually more memory consumed by the entire cache by having another object reference. (hope that makes sense?)

Reply:
Yes, makes sense.  As you say, it does simplify things too.  Will take a look at updating to this.